state/accountsDB.go

```diff
@@ -297,12 +298,12 @@ func (adb *AccountsDB) SaveAccount(account vmcommon.AccountHandler) error {
 	return nil
 }
 
-func (adb *AccountsDB) saveCodeAndDataTrie(oldAcc, newAcc vmcommon.AccountHandler) ([]stateChanges.DataTrieChange, error) {
+func (adb *AccountsDB) saveCodeAndDataTrie(oldAcc, newAcc vmcommon.AccountHandler) ([]*stateChange.DataTrieChange, error) {
 	baseNewAcc, newAccOk := newAcc.(baseAccountHandler)
 	baseOldAccount, _ := oldAcc.(baseAccountHandler)
 
 	if !newAccOk {
-		return nil, nil
+		return make([]*stateChange.DataTrieChange, 0), nil
```

Comment:
Is this required because of a panic?

Reply:
`TestMetaProcessor_CommitBlockOkValsShouldWork` yes, this test would fail with the previous implementation.

Reply:
@BeniaminDrasovean can you check if this change would be problematic?
do we need to change the test instead?

Reply:
It's ok both ways.